pkg/sink/sink.go

```diff
@@ -64,8 +64,14 @@ type Response struct {
 	Namespace string `json:"namespace,omitempty"`
 	// EventID is a uniqueID that gets assigned to each incoming request
 	EventID string `json:"eventID,omitempty"`
+	// Error message contains any error that occur during processing event
+	Error string `json:"error,omitempty"`
```

Comment:
I agree with @dibyom on providing a structured per-trigger response type, instead of trying to infer individual trigger statuses from the overall result.

The same way it's misleading that a 2xx response results in an error, I'm worried that bubbling up more per-trigger errors at the event level misleads people into thinking nothing happened, when several triggers could have succeeded. Unlike the existing 2xx behavior, this might have more severe consequences (e.g. spawning many more Runs than intended).

I think https://google.aip.dev/193#partial-errors would be a great model to follow / draw inspiration from here (particularly the use of long running operations and returning subresource errors in metadata).